huaweicloud/services/live/resource_huaweicloud_live_origin_pull_configuration.go

```diff
@@ -0,0 +1,250 @@
+	validationPath := client.Endpoint + validationHttpUrl
+	validationPath = strings.ReplaceAll(validationPath, "{project_id}", client.ProjectID)
+
+	validationOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildOriginPullConfigurationBodyParams(d)),
+	}
+
+	_, err := client.Request("PUT", validationPath, &validationOpt)
+	return err
+}
+
+func resourceOriginPullConfigurationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	err = updateOriginPullConfiguration(client, d)
+	if err != nil {
+		return diag.Errorf("error creating Live origin pull configuration: %s", err)
+	}
+
+	resourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+
+	d.SetId(resourceId)
+
+	return resourceOriginPullConfigurationRead(ctx, d, meta)
+}
+
+func ReadOriginPullConfiguration(client *golangsdk.ServiceClient, domainName string) (interface{}, error) {
+	getPath := client.Endpoint + "v1/{project_id}/domain/pull-sources"
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath += fmt.Sprintf("?play_domain=%s", domainName)
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		// When the `domain_name` does not exist, calling the query API will return a `400` status code.
+		return nil, common.ConvertExpected400ErrInto404Err(err, "error_code", domainNameNotExistsCode)
+	}
+
+	return utils.FlattenResponse(getResp)
+}
+
+func resourceOriginPullConfigurationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg        = meta.(*config.Config)
+		region     = cfg.GetRegion(d)
+		domainName = d.Get("domain_name").(string)
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	getRespBody, err := ReadOriginPullConfiguration(client, domainName)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Live origin pull configuration")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("domain_name", domainName),
+		d.Set("source_type", utils.PathSearch("source_type", getRespBody, nil)),
+		d.Set("sources", flattenSourcesOrSourcesIp(utils.PathSearch("sources", getRespBody, nil))),
+		d.Set("sources_ip", flattenSourcesOrSourcesIp(utils.PathSearch("sources_ip", getRespBody, nil))),
+		d.Set("source_port", utils.PathSearch("source_port", getRespBody, float64(0))),
+		d.Set("scheme", utils.PathSearch("scheme", getRespBody, nil)),
+		d.Set("additional_args", flattenAdditionalArgs(utils.PathSearch("additional_args", getRespBody, nil))),
```

Comment:
Backfilling map type attributes is not recommended, as it may trigger incompatible changes.

Reply:
done.